src/downloader.ts

```diff
@@ -82,14 +85,21 @@ export class Downloader {
 
         // download file and pipe to stream
         if (parsedURL.protocol === 'https:') {
-            Downloader.https.get(fileURL, response => {
+            const request = Downloader.https.get(fileURL, response => {
                 callback(undefined, response);
             });
+            request.setTimeout(Downloader.timeout, response => {
+                let err : Error = new RequestTimeoutError('Request for file:'+parsedURL.pathname+' was timed out after '+Downloader.timeout+' ms');
+                callback(err,undefined);
+            });
         } else if (parsedURL.protocol === 'http:') {
-            Downloader.http.get(fileURL, response => {
+            const request = Downloader.http.get(fileURL, response => {
                 callback(undefined, response);
             });
+            request.setTimeout(Downloader.timeout, response => {
```

Comment:
Hm, true. I didn't think about that. The signature should look like this then to ensure backwards compatibly
```
getResponse(fileURL : string, callback? : (err? : Error, resp? : IncomingMessage, timeout? : number) => void)
```